App/Sagas/TextileSagas.js

```diff
@@ -239,24 +239,52 @@ export function * shareImage ({thread, hash, caption}) {
 
 export function * photosTask () {
   try {
-    const photos = yield call(queryPhotos)
-    for (const photo of photos.reverse()) {
+    const camera = yield select(TextileSelectors.camera)
+    const processed = camera && camera.processed ? camera.processed : []
+
+    let allPhotos = yield call(getAllPhotos)
+    if (camera === undefined || camera.processed === undefined || processed.length === 0) {
```

Comment:
This can just be `if (processed.length === 0) {...}` because all the `undefined` checks are covered on line 243 above.

Reply:
done